L1Trigger/L1THGCal/src/backend/HGCalShowerShape.cc

```diff
@@ -87,6 +87,8 @@ int HGCalShowerShape::coreShowerLength(const l1t::HGCalMulticluster& c3d,
     if (!pass(*id_cluster.second, c3d))
       continue;
     unsigned layer = triggerGeometry.triggerLayer(id_cluster.second->detId());
+    if (triggerTools_.isNose(id_cluster.second->detId()))
+      nlayers = triggerTools_.layers(ForwardSubdetector::HFNose);
```

Comment:
the possible overlap is around eta 3
the idea is to have this done at correlator level and not at the calo level.
So we keep separated the FW and SW for HFnose and HGC.
Once the 11_0 is out the overlap region can be better studied.

Reply:
Right now  `nLayers`, which should be constant for a given cluster, is modified when a Nose TC is found in the cluster. Which means it will break the output of this function when a cluster is both in the HGCAL and the nose. This should be fixed.

Reply:
fixed, but putting back the original default is noNose

Reply:
Now I think the `nlayers` value is correct. Still to be fixed in the other method `percentileLayer`

Reply:
fixed the percentileLayer as well